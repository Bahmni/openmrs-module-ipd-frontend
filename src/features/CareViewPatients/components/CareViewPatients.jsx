import React from "react";
import "../styles/CareViewPatients.scss";
import { Accordion, AccordionItem, Dropdown } from "carbon-components-react";

export const CareViewPatients = () => {
  return (
    <div className="care-view-patients-container">
      <div className="care-view-patients">
        <div className="task-type">
          <Dropdown
            id="default"
            label="Dropdown menu options"
            items={["All tasks", "Pending", "Done"]}
            itemToString={(item) => (item ? item : "")}
            selectedItem={"All tasks"}
          />
        </div>
        <Accordion className="patient-section">
          <AccordionItem title="Patient 1">Patient 1</AccordionItem>
          <AccordionItem title="Patient 2">Patient 2</AccordionItem>
        </Accordion>
      </div>
    </div>
  );
};
