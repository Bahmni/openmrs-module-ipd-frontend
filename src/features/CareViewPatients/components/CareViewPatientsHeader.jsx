import React from "react";
import PropTypes from "prop-types";
import "../styles/CareViewPatientsHeader.scss";
import { IPD_WARD_SEARCH_PLACEHOLDER_TEXT } from "../../../constants";
import { Dropdown, Search } from "carbon-components-react";

export const CareViewPatientsHeader = (props) => {
  const { handleKeyPress, handleClear, searchValue, updateSearchValue } = props;

  const handleSearchOnChange = (e) => {
    updateSearchValue(e.target.value);
  };

  return (
    <div className="task-type">
      <Search
        placeholder={IPD_WARD_SEARCH_PLACEHOLDER_TEXT}
        size="lg"
        id="ipd-ward-search"
        onChange={handleSearchOnChange}
        value={searchValue}
        onKeyDown={handleKeyPress}
        onClear={handleClear}
        className="ward-patient-search"
        labelText="Patient Search"
        light={true}
      />
      <Dropdown
        id="default"
        label="Dropdown menu options"
        items={["All tasks", "Pending", "Done"]}
        itemToString={(item) => (item ? item : "")}
        selectedItem={"All tasks"}
        light={true}
      />
    </div>
  );
};

CareViewPatientsHeader.propTypes = {
  handleKeyPress: PropTypes.func,
  handleClear: PropTypes.func,
  searchValue: PropTypes.string,
  updateSearchValue: PropTypes.func,
};
