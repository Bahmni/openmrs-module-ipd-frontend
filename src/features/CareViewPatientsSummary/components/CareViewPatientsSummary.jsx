import React, { useEffect, useState } from "react";
import propTypes from "prop-types";
import "../styles/CareViewPatientsSummary.scss";
import { HospitalBed16 } from "@carbon/icons-react";
import { Link } from "carbon-components-react";
import { FormattedMessage } from "react-intl";
import { getSlotsForPatientsAndTime } from "../../CareViewSummary/utils/CareViewSummary";

export const CareViewPatientsSummary = (props) => {
  const [setSlotDetails] = useState([]);
  const { patientsSummary } = props;

  const fetchSlots = async (patientUuids, currentTime, endTime) => {
    const response = await getSlotsForPatientsAndTime(
      patientUuids,
      currentTime,
      endTime
    );
    setSlotDetails(response.data);
  };

  useEffect(() => {
    if (patientsSummary.length > 0) {
      const patientUuids = patientsSummary.map(
        (patient) => patient.patientDetails.uuid
      );
      const currentTime = Math.floor(Date.now() / 1000);

      const getTimeInFuture = (offsetHours) => {
        const futureTime = new Date();
        futureTime.setHours(futureTime.getHours() + offsetHours);
        const futureTimeUnix = Math.floor(futureTime.getTime() / 1000);
        return futureTimeUnix.toString();
      };
      const endTime = getTimeInFuture(2);
      fetchSlots(patientUuids, currentTime, endTime);
    }
  }, [patientsSummary]);

  return (
    <div>
      <table className={"care-view-patient-table"}>
        <tbody>
          {patientsSummary.map((patientSummary, idx) => {
            const { patientDetails, bedDetails } = patientSummary;
            const { person } = patientDetails;
            return (
              <tr key={idx} className={"patient-row-container"}>
                <td className={"patient-details-container"}>
                  <div className={"care-view-patient-details"}>
                    <div>
                      <HospitalBed16 />
                      &nbsp;
                      <span>{bedDetails.bedNumber}</span>&nbsp;
                      <Link href={"#"} inline={true}>
                        <span>{patientDetails.display.split(" ")[0]}</span>
                      </Link>
                    </div>
                    <div>
                      <FormattedMessage
                        id={"PATIENT"}
                        defaultMessage={"Patient"}
                      />
                      : <span>{person.display}</span>&nbsp;(
                      <span>{person.gender}</span>)&nbsp;.
                      <span>{person.age}</span>
                      <FormattedMessage
                        id={"AGE_YEARS_LABEL"}
                        defaultMessage={"yrs"}
                      />
                    </div>
                  </div>
                </td>
                <td style={{ minWidth: "650px" }}></td>
                <td style={{ minWidth: "650px" }}></td>
              </tr>
            );
          })}
        </tbody>
      </table>
    </div>
  );
};

CareViewPatientsSummary.propTypes = {
  patientsSummary: propTypes.array,
};
