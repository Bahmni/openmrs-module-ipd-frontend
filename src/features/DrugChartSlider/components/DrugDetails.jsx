import React from "react";
import { TextInput } from "carbon-components-react";
import {
  NumberInputCarbon,
  DropdownCarbon,
  DatePickerCarbon,
} from "bahmni-carbon-ui";
import PropTypes from "prop-types";
import moment from "moment";

export const DrugDetails = ({ hostData }) => {
  return (
    <>
      <TextInput
        id="drug-name"
        className="drug-name"
        type="text"
        value={hostData?.drugOrder?.drugOrder?.drug?.name}
        labelText="Drug Name"
        disabled
      />
      <div className="inline-field">
        <div className="dose-field-with-units">
          <NumberInputCarbon
            id={"Dropdown"}
            onChange={() => {}}
            style={{ width: "50%" }}
            label={"Dose"}
            value={hostData?.drugOrder?.uniformDosingType?.dose}
            isDisabled={true}
          />
          <DropdownCarbon
            id={"Dropdown"}
            onChange={() => {}}
            titleText={" "}
            style={{ paddingLeft: "10px" }}
            selectedValue={hostData?.drugOrder?.uniformDosingType?.doseUnits}
            options={[]}
            isDisabled={true}
          />
        </div>
        <div className="route">
          <DropdownCarbon
            id={"Dropdown"}
            onChange={() => {}}
            titleText={"Route"}
            selectedValue={hostData?.drugOrder?.route}
            options={[]}
            isDisabled={true}
          />
        </div>
      </div>
      <div className="inline-field">
        <div className="duration-field-with-units">
          <NumberInputCarbon
            id={"Dropdown"}
            onChange={() => {}}
            label={"Duration"}
            value={hostData?.drugOrder?.drugOrder?.duration}
            isDisabled={true}
          />
          <DropdownCarbon
            id={"Dropdown"}
            onChange={() => {}}
            titleText={" "}
            selectedValue={hostData?.drugOrder?.drugOrder?.durationUnits}
            options={[]}
            isDisabled={true}
          />
        </div>
        <DatePickerCarbon
          id={"Dropdown"}
          onChange={() => {}}
          titleText={"Start Date"}
          title={"Start Date"}
          value={moment(hostData?.drugOrder?.drugOrder?.scheduledDate).format(
            "MM/DD/YYYY"
          )}
          isDisabled={true}
        />
      </div>
      <div className="frequency">
        <DropdownCarbon
          id={"DropdownFrequency"}
          onChange={() => {}}
          titleText={"Frequency"}
          selectedValue={hostData?.drugOrder?.uniformDosingType?.frequency}
          options={[]}
          isDisabled={true}
        />
      </div>
    </>
  );
};

DrugDetails.propTypes = {
  hostData: PropTypes.shape({
    drugOrder: PropTypes.shape({
      drug: PropTypes.shape({
        name: PropTypes.string,
      }),
      uniformDosingType: PropTypes.shape({
        dose: PropTypes.string,
        doseUnits: PropTypes.string,
        frequency: PropTypes.string,
      }),
      route: PropTypes.string,
      duration: PropTypes.string,
      durationUnit: PropTypes.string,
      scheduledDate: PropTypes.string,
    }),
  }),
};
