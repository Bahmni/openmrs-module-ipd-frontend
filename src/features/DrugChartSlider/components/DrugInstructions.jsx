import React from "react";
import { TextArea } from "carbon-components-react";
import PropTypes from "prop-types";

export const DrugInstructions = ({ hostData }) => {
  return (
    <>
      <div className="instructions">
        <TextArea
          className="instruction"
          readOnly
          type="text"
          rows={1}
          value={hostData?.drugOrder?.instructions}
          labelText="Instruction"
          disabled
        />
      </div>
      <div className="additional-instructions">
        <TextArea
          className="additional-instruction"
          readOnly
          type="text"
          rows={1}
          value={hostData?.drugOrder?.additionalInstructions}
          labelText="Additional Instruction"
          disabled
        />
      </div>
    </>
  );
};

DrugInstructions.propTypes = {
  hostData: PropTypes.shape({
    drugOrder: PropTypes.shape({
      instructions: PropTypes.string,
      additionalInstructions: PropTypes.string,
    }),
  }),
};
