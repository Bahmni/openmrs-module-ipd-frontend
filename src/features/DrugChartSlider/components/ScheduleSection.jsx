import React from "react";
import PropTypes from "prop-types";
import {
  invalidTimeText12Hour,
  invalidTimeText24Hour,
} from "../utils/DrugChartSliderUtils";
import { FormattedMessage } from "react-intl";
import { TimePicker, TimePicker24Hour, Title } from "bahmni-carbon-ui";

export const ScheduleSection = ({
  enableSchedule,
  firstDaySlotsMissed,
  firstDaySchedules,
  schedules,
  finalDaySchedules,
  handleFirstDaySchedule,
  handleSubsequentDaySchedule,
  handleFinalDaySchedule,
  showFirstDayScheduleOrderWarning,
  showEmptyFirstDayScheduleWarning,
  showFirstDaySchedulePassedWarning,
  showScheduleOrderWarning,
  showEmptyScheduleWarning,
  showFinalDayScheduleOrderWarning,
  showEmptyFinalDayScheduleWarning,
  showSchedulePassedWarning,
  enable24HourTimers,
}) => {
  return (
    <>
      {enableSchedule && firstDaySlotsMissed > 0 && (
        <div>
          <div className="schedule-section">
            <Title text="Schedule time (start date)" isRequired={true} />
            <div className="inline-field" id="schedule">
              {Array.from(
                { length: enableSchedule?.frequencyPerDay },
                (_, index) =>
                  enable24HourTimers ? (
                    <div className="schedule-time" key={index}>
                      <TimePicker24Hour
                        key={index}
                        id={`schedule-${index}`}
                        defaultTime={firstDaySchedules[index]}
                        onChange={(time) => {
                          handleFirstDaySchedule(time, index);
                        }}
                        labelText=" "
                        width="70%"
                        invalidText={invalidTimeText24Hour}
                        isDisabled={firstDaySchedules[index] == "hh:mm"}
                      />
                    </div>
                  ) : (
                    <div className="schedule-time" key={index}>
                      <TimePicker
                        key={index}
                        labelText=" "
                        defaultTime={firstDaySchedules[index]}
                        onChange={(time) => {
                          handleFirstDaySchedule(time, index);
                        }}
                        id={`schedule-${index}`}
                        isDisabled={firstDaySchedules[index] == "hh:mm"}
                        invalidText={invalidTimeText12Hour}
                      />
                    </div>
                  )
              )}
            </div>
            {showFirstDayScheduleOrderWarning && (
              <p className="time-error">
                <FormattedMessage id="DRUG_CHART_MODAL_SCHEDULE_ORDER_WARNING"></FormattedMessage>
              </p>
            )}
            {showEmptyFirstDayScheduleWarning && (
              <p className="time-error">
                <FormattedMessage id="DRUG_CHART_MODAL_EMPTY_SCHEDULE_WARNING"></FormattedMessage>
              </p>
            )}
            {showFirstDaySchedulePassedWarning.some(
              (showSchedulePassed) => showSchedulePassed === true
            ) && (
              <p className="time-warning">
                <FormattedMessage id="DRUG_CHART_MODAL_SCHEDULE_PASSED_WARNING"></FormattedMessage>
              </p>
            )}
          </div>
          <div className="schedule-section">
            <Title text="Schedule time (subsequent)" isRequired={true} />
            <div className="inline-field" id="schedule">
              {Array.from(
                { length: enableSchedule?.frequencyPerDay },
                (_, index) =>
                  enable24HourTimers ? (
                    <div className="schedule-time" key={index}>
                      <TimePicker24Hour
                        key={index}
                        id={`schedule-${index}`}
                        defaultTime={schedules[index]}
                        onChange={(time) => {
                          handleSubsequentDaySchedule(time, index);
                        }}
                        labelText=" "
                        width="70%"
                        invalidText={invalidTimeText24Hour}
                      />
                    </div>
                  ) : (
                    <div className="schedule-time" key={index}>
                      <TimePicker
                        key={index}
                        labelText=" "
                        defaultTime={schedules[index]}
                        onChange={(time) => {
                          handleSubsequentDaySchedule(time, index);
                        }}
                        id={`schedule-${index}`}
                        invalidText={invalidTimeText12Hour}
                      />
                    </div>
                  )
              )}
            </div>
            {showScheduleOrderWarning && (
              <p className="time-error">
                <FormattedMessage id="DRUG_CHART_MODAL_SCHEDULE_ORDER_WARNING"></FormattedMessage>
              </p>
            )}
            {showEmptyScheduleWarning && (
              <p className="time-error">
                <FormattedMessage id="DRUG_CHART_MODAL_EMPTY_SCHEDULE_WARNING"></FormattedMessage>
              </p>
            )}
          </div>
          <div className="schedule-section">
            <Title text="Schedule time (remainder)" isRequired={true} />
            <div className="inline-field" id="schedule">
              {Array.from(
                {
                  length: firstDaySlotsMissed,
                },
                (_, index) =>
                  enable24HourTimers ? (
                    <div
                      className={
                        enableSchedule?.frequencyPerDay == 4
                          ? "schedule-time-remainder"
                          : "schedule-time"
                      }
                      key={index}
                    >
                      <TimePicker24Hour
                        key={index}
                        id={`schedule-${index}`}
                        defaultTime={finalDaySchedules[index]}
                        onChange={(time) => {
                          handleFinalDaySchedule(time, index);
                        }}
                        labelText=" "
                        width="70%"
                        invalidText={invalidTimeText24Hour}
                      />
                    </div>
                  ) : (
                    <div
                      className={
                        enableSchedule?.frequencyPerDay == 4
                          ? "schedule-time-remainder"
                          : "schedule-time"
                      }
                      key={index}
                    >
                      <TimePicker
                        key={index}
                        labelText=" "
                        defaultTime={finalDaySchedules[index]}
                        onChange={(time) => {
                          handleFinalDaySchedule(time, index);
                        }}
                        id={`schedule-${index}`}
                        invalidText={invalidTimeText12Hour}
                      />
                    </div>
                  )
              )}
            </div>
          </div>
          {showFinalDayScheduleOrderWarning && (
            <p className="time-error">
              <FormattedMessage id="DRUG_CHART_MODAL_SCHEDULE_ORDER_WARNING"></FormattedMessage>
            </p>
          )}
          {showEmptyFinalDayScheduleWarning && (
            <p className="time-error">
              <FormattedMessage id="DRUG_CHART_MODAL_EMPTY_SCHEDULE_WARNING"></FormattedMessage>
            </p>
          )}
        </div>
      )}
      {enableSchedule && firstDaySlotsMissed == 0 && (
        <div>
          <div className="schedule-section">
            <Title text="Schedule(s)" isRequired={true} />
            <div className="inline-field" id="schedule">
              {Array.from(
                { length: enableSchedule?.frequencyPerDay },
                (_, index) =>
                  enable24HourTimers ? (
                    <div className="schedule-time" key={index}>
                      <TimePicker24Hour
                        key={index}
                        id={`schedule-${index}`}
                        defaultTime={schedules[index]}
                        onChange={(time) => {
                          handleSubsequentDaySchedule(time, index);
                        }}
                        labelText=" "
                        width="70%"
                        invalidText={invalidTimeText24Hour}
                      />
                    </div>
                  ) : (
                    <div className="schedule-time" key={index}>
                      <TimePicker
                        key={index}
                        labelText=" "
                        defaultTime={schedules[index]}
                        onChange={(time) => {
                          handleSubsequentDaySchedule(time, index);
                        }}
                        id={`schedule-${index}`}
                        invalidText={invalidTimeText12Hour}
                      />
                    </div>
                  )
              )}
            </div>
            {showScheduleOrderWarning && (
              <p className="time-error">
                <FormattedMessage id="DRUG_CHART_MODAL_SCHEDULE_ORDER_WARNING"></FormattedMessage>
              </p>
            )}
            {showEmptyScheduleWarning && (
              <p className="time-error">
                <FormattedMessage id="DRUG_CHART_MODAL_EMPTY_SCHEDULE_WARNING"></FormattedMessage>
              </p>
            )}
            {showSchedulePassedWarning.some(
              (showSchedulePassed) => showSchedulePassed === true
            ) && (
              <p className="time-warning">
                <FormattedMessage id="DRUG_CHART_MODAL_SCHEDULE_PASSED_WARNING"></FormattedMessage>
              </p>
            )}
          </div>
        </div>
      )}
    </>
  );
};

ScheduleSection.propTypes = {
  enableSchedule: PropTypes.shape({
    frequencyPerDay: PropTypes.number.isRequired,
  }),
  firstDaySlotsMissed: PropTypes.number.isRequired,
  firstDaySchedules: PropTypes.array.isRequired,
  schedules: PropTypes.array.isRequired,
  finalDaySchedules: PropTypes.array.isRequired,
  handleFirstDaySchedule: PropTypes.func.isRequired,
  handleSubsequentDaySchedule: PropTypes.func.isRequired,
  handleFinalDaySchedule: PropTypes.func.isRequired,
  showFirstDayScheduleOrderWarning: PropTypes.bool.isRequired,
  showEmptyFirstDayScheduleWarning: PropTypes.bool.isRequired,
  showFirstDaySchedulePassedWarning: PropTypes.array.isRequired,
  showScheduleOrderWarning: PropTypes.bool.isRequired,
  showEmptyScheduleWarning: PropTypes.bool.isRequired,
  showFinalDayScheduleOrderWarning: PropTypes.bool.isRequired,
  showEmptyFinalDayScheduleWarning: PropTypes.bool.isRequired,
  showSchedulePassedWarning: PropTypes.array.isRequired,
  enable24HourTimers: PropTypes.bool.isRequired,
};
