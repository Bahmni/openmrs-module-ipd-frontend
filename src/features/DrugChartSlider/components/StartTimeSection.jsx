// StartTimeSection.js
import React from "react";
import { FormattedMessage } from "react-intl";
import { TimePicker, TimePicker24Hour } from "bahmni-carbon-ui";
import PropTypes from "prop-types";
import {
  invalidTimeText12Hour,
  invalidTimeText24Hour,
} from "../utils/DrugChartSliderUtils";

export const StartTimeSection = ({
  startTime,
  handleStartTime,
  showEmptyStartTimeWarning,
  showStartTimeBeyondNextDoseWarning,
  showStartTimePassedWarning,
  enable24HourTimers,
}) => {
  return (
    <div className="start-time">
      {enable24HourTimers ? (
        <>
          <TimePicker24Hour
            data-modal-primary-focus
            labelText={"Start Time"}
            id={"start-time"}
            onChange={(time) => handleStartTime(time)}
            isRequired={true}
            defaultTime={startTime}
            width="80%"
            invalidText={invalidTimeText24Hour}
          />
          {showStartTimePassedWarning && (
            <p className="time-warning">
              <FormattedMessage id="DRUG_CHART_MODAL_START_TIME_PASSED"></FormattedMessage>
            </p>
          )}
          {showStartTimeBeyondNextDoseWarning && (
            <p className="time-warning">
              <FormattedMessage id="DRUG_CHART_MODAL_START_TIME_BEYOND_NEXT_DOSE"></FormattedMessage>
            </p>
          )}
          {showEmptyStartTimeWarning && (
            <p className="time-error">
              <FormattedMessage id="DRUG_CHART_MODAL_EMPTY_START_TIME_WARNING"></FormattedMessage>
            </p>
          )}
        </>
      ) : (
        <>
          <TimePicker
            id={"start-time"}
            onChange={(time) => handleStartTime(time)}
            defaultTime={startTime}
            labelText={"Start Time"}
            isRequired={true}
            invalidText={invalidTimeText12Hour}
          />
          {showStartTimePassedWarning && (
            <p className="time-warning">
              <FormattedMessage id="DRUG_CHART_MODAL_START_TIME_PASSED"></FormattedMessage>
            </p>
          )}
          {showStartTimeBeyondNextDoseWarning && (
            <p className="time-warning">
              <FormattedMessage id="DRUG_CHART_MODAL_START_TIME_BEYOND_NEXT_DOSE"></FormattedMessage>
            </p>
          )}
          {showEmptyStartTimeWarning && (
            <p className="time-error">
              <FormattedMessage id="DRUG_CHART_MODAL_EMPTY_START_TIME_WARNING"></FormattedMessage>
            </p>
          )}
        </>
      )}
    </div>
  );
};

StartTimeSection.propTypes = {
  startTime: PropTypes.string.isRequired,
  handleStartTime: PropTypes.func.isRequired,
  showEmptyStartTimeWarning: PropTypes.bool.isRequired,
  showStartTimeBeyondNextDoseWarning: PropTypes.bool.isRequired,
  showStartTimePassedWarning: PropTypes.bool.isRequired,
  enable24HourTimers: PropTypes.bool.isRequired,
};
