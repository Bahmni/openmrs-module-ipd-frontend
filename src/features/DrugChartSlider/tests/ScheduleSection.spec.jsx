import React from "react";
import { render, waitFor } from "@testing-library/react";
import { ScheduleSection } from "../components/ScheduleSection"; // Replace 'YourComponent' with the actual file name

describe("ScheduleSection", () => {
  const props = {
    enableSchedule: { frequencyPerDay: 3 },
    firstDaySlotsMissed: 2,
    firstDaySchedules: ["08:00", "12:00", "16:00"],
    schedules: ["09:00", "13:00", "17:00"],
    finalDaySchedules: ["10:00", "14:00"],
    handleFirstDaySchedule: jest.fn(),
    handleSubsequentDaySchedule: jest.fn(),
    handleFinalDaySchedule: jest.fn(),
    showFirstDayScheduleOrderWarning: false,
    showEmptyFirstDayScheduleWarning: false,
    showFirstDaySchedulePassedWarning: [false, false, false],
    showScheduleOrderWarning: false,
    showEmptyScheduleWarning: false,
    showFinalDayScheduleOrderWarning: false,
    showEmptyFinalDayScheduleWarning: false,
    showSchedulePassedWarning: [false, false, false],
    enable24HourTimers: true,
  };

  it("renders ScheduleSection component when firstday slots are missed", async () => {
    const { container } = render(<ScheduleSection {...props} />);

    await waitFor(() => {
      expect(container).toMatchSnapshot();
    });
  });
  it("renders ScheduleSection component when firstday slots  missed is 0", async () => {
    const { container } = render(
      <ScheduleSection {...props} firstDaySlotsMissed={0} />
    );

    await waitFor(() => {
      expect(container).toMatchSnapshot();
    });
  });
});
