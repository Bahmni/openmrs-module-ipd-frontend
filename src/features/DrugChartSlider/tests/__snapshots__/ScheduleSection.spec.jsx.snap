// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ScheduleSection renders ScheduleSection component when firstday slots  missed is 0 1`] = `
<div>
  <div>
    <div
      class="schedule-section"
    >
      <div
        class="titleText"
      >
        <span>
          Schedule(s) 
        </span>
        <span
          class="required"
        >
          *
        </span>
      </div>
      <div
        class="inline-field"
        id="schedule"
      >
        <div
          class="schedule-time"
        >
          <div
            class="bx--form-item"
          >
            <label
              class="bx--label"
              for="time-selector"
            >
              <div
                class="titleText"
              >
                <span>
                    
                </span>
              </div>
            </label>
            <div
              class="bx--time-picker"
            >
              <div
                class="bx--time-picker__input"
              >
                <input
                  autocomplete="off"
                  class="bx--time-picker__input-field bx--text-input"
                  id="time-selector"
                  maxlength="5"
                  pattern="(1[012]|[1-9]):[0-5][0-9](\\s)?"
                  placeholder="hh:mm"
                  style="width: 70%; padding: 0px 0px 0px 1rem;"
                  type="text"
                  value="09:00"
                />
              </div>
            </div>
          </div>
        </div>
        <div
          class="schedule-time"
        >
          <div
            class="bx--form-item"
          >
            <label
              class="bx--label"
              for="time-selector"
            >
              <div
                class="titleText"
              >
                <span>
                    
                </span>
              </div>
            </label>
            <div
              class="bx--time-picker"
            >
              <div
                class="bx--time-picker__input"
              >
                <input
                  autocomplete="off"
                  class="bx--time-picker__input-field bx--text-input"
                  id="time-selector"
                  maxlength="5"
                  pattern="(1[012]|[1-9]):[0-5][0-9](\\s)?"
                  placeholder="hh:mm"
                  style="width: 70%; padding: 0px 0px 0px 1rem;"
                  type="text"
                  value="13:00"
                />
              </div>
            </div>
          </div>
        </div>
        <div
          class="schedule-time"
        >
          <div
            class="bx--form-item"
          >
            <label
              class="bx--label"
              for="time-selector"
            >
              <div
                class="titleText"
              >
                <span>
                    
                </span>
              </div>
            </label>
            <div
              class="bx--time-picker"
            >
              <div
                class="bx--time-picker__input"
              >
                <input
                  autocomplete="off"
                  class="bx--time-picker__input-field bx--text-input"
                  id="time-selector"
                  maxlength="5"
                  pattern="(1[012]|[1-9]):[0-5][0-9](\\s)?"
                  placeholder="hh:mm"
                  style="width: 70%; padding: 0px 0px 0px 1rem;"
                  type="text"
                  value="17:00"
                />
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`ScheduleSection renders ScheduleSection component when firstday slots are missed 1`] = `
<div>
  <div>
    <div
      class="schedule-section"
    >
      <div
        class="titleText"
      >
        <span>
          Schedule time (start date) 
        </span>
        <span
          class="required"
        >
          *
        </span>
      </div>
      <div
        class="inline-field"
        id="schedule"
      >
        <div
          class="schedule-time"
        >
          <div
            class="bx--form-item"
          >
            <label
              class="bx--label"
              for="time-selector"
            >
              <div
                class="titleText"
              >
                <span>
                    
                </span>
              </div>
            </label>
            <div
              class="bx--time-picker"
            >
              <div
                class="bx--time-picker__input"
              >
                <input
                  autocomplete="off"
                  class="bx--time-picker__input-field bx--text-input"
                  id="time-selector"
                  maxlength="5"
                  pattern="(1[012]|[1-9]):[0-5][0-9](\\s)?"
                  placeholder="hh:mm"
                  style="width: 70%; padding: 0px 0px 0px 1rem;"
                  type="text"
                  value="08:00"
                />
              </div>
            </div>
          </div>
        </div>
        <div
          class="schedule-time"
        >
          <div
            class="bx--form-item"
          >
            <label
              class="bx--label"
              for="time-selector"
            >
              <div
                class="titleText"
              >
                <span>
                    
                </span>
              </div>
            </label>
            <div
              class="bx--time-picker"
            >
              <div
                class="bx--time-picker__input"
              >
                <input
                  autocomplete="off"
                  class="bx--time-picker__input-field bx--text-input"
                  id="time-selector"
                  maxlength="5"
                  pattern="(1[012]|[1-9]):[0-5][0-9](\\s)?"
                  placeholder="hh:mm"
                  style="width: 70%; padding: 0px 0px 0px 1rem;"
                  type="text"
                  value="12:00"
                />
              </div>
            </div>
          </div>
        </div>
        <div
          class="schedule-time"
        >
          <div
            class="bx--form-item"
          >
            <label
              class="bx--label"
              for="time-selector"
            >
              <div
                class="titleText"
              >
                <span>
                    
                </span>
              </div>
            </label>
            <div
              class="bx--time-picker"
            >
              <div
                class="bx--time-picker__input"
              >
                <input
                  autocomplete="off"
                  class="bx--time-picker__input-field bx--text-input"
                  id="time-selector"
                  maxlength="5"
                  pattern="(1[012]|[1-9]):[0-5][0-9](\\s)?"
                  placeholder="hh:mm"
                  style="width: 70%; padding: 0px 0px 0px 1rem;"
                  type="text"
                  value="16:00"
                />
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
    <div
      class="schedule-section"
    >
      <div
        class="titleText"
      >
        <span>
          Schedule time (subsequent) 
        </span>
        <span
          class="required"
        >
          *
        </span>
      </div>
      <div
        class="inline-field"
        id="schedule"
      >
        <div
          class="schedule-time"
        >
          <div
            class="bx--form-item"
          >
            <label
              class="bx--label"
              for="time-selector"
            >
              <div
                class="titleText"
              >
                <span>
                    
                </span>
              </div>
            </label>
            <div
              class="bx--time-picker"
            >
              <div
                class="bx--time-picker__input"
              >
                <input
                  autocomplete="off"
                  class="bx--time-picker__input-field bx--text-input"
                  id="time-selector"
                  maxlength="5"
                  pattern="(1[012]|[1-9]):[0-5][0-9](\\s)?"
                  placeholder="hh:mm"
                  style="width: 70%; padding: 0px 0px 0px 1rem;"
                  type="text"
                  value="09:00"
                />
              </div>
            </div>
          </div>
        </div>
        <div
          class="schedule-time"
        >
          <div
            class="bx--form-item"
          >
            <label
              class="bx--label"
              for="time-selector"
            >
              <div
                class="titleText"
              >
                <span>
                    
                </span>
              </div>
            </label>
            <div
              class="bx--time-picker"
            >
              <div
                class="bx--time-picker__input"
              >
                <input
                  autocomplete="off"
                  class="bx--time-picker__input-field bx--text-input"
                  id="time-selector"
                  maxlength="5"
                  pattern="(1[012]|[1-9]):[0-5][0-9](\\s)?"
                  placeholder="hh:mm"
                  style="width: 70%; padding: 0px 0px 0px 1rem;"
                  type="text"
                  value="13:00"
                />
              </div>
            </div>
          </div>
        </div>
        <div
          class="schedule-time"
        >
          <div
            class="bx--form-item"
          >
            <label
              class="bx--label"
              for="time-selector"
            >
              <div
                class="titleText"
              >
                <span>
                    
                </span>
              </div>
            </label>
            <div
              class="bx--time-picker"
            >
              <div
                class="bx--time-picker__input"
              >
                <input
                  autocomplete="off"
                  class="bx--time-picker__input-field bx--text-input"
                  id="time-selector"
                  maxlength="5"
                  pattern="(1[012]|[1-9]):[0-5][0-9](\\s)?"
                  placeholder="hh:mm"
                  style="width: 70%; padding: 0px 0px 0px 1rem;"
                  type="text"
                  value="17:00"
                />
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
    <div
      class="schedule-section"
    >
      <div
        class="titleText"
      >
        <span>
          Schedule time (remainder) 
        </span>
        <span
          class="required"
        >
          *
        </span>
      </div>
      <div
        class="inline-field"
        id="schedule"
      >
        <div
          class="schedule-time"
        >
          <div
            class="bx--form-item"
          >
            <label
              class="bx--label"
              for="time-selector"
            >
              <div
                class="titleText"
              >
                <span>
                    
                </span>
              </div>
            </label>
            <div
              class="bx--time-picker"
            >
              <div
                class="bx--time-picker__input"
              >
                <input
                  autocomplete="off"
                  class="bx--time-picker__input-field bx--text-input"
                  id="time-selector"
                  maxlength="5"
                  pattern="(1[012]|[1-9]):[0-5][0-9](\\s)?"
                  placeholder="hh:mm"
                  style="width: 70%; padding: 0px 0px 0px 1rem;"
                  type="text"
                  value="10:00"
                />
              </div>
            </div>
          </div>
        </div>
        <div
          class="schedule-time"
        >
          <div
            class="bx--form-item"
          >
            <label
              class="bx--label"
              for="time-selector"
            >
              <div
                class="titleText"
              >
                <span>
                    
                </span>
              </div>
            </label>
            <div
              class="bx--time-picker"
            >
              <div
                class="bx--time-picker__input"
              >
                <input
                  autocomplete="off"
                  class="bx--time-picker__input-field bx--text-input"
                  id="time-selector"
                  maxlength="5"
                  pattern="(1[012]|[1-9]):[0-5][0-9](\\s)?"
                  placeholder="hh:mm"
                  style="width: 70%; padding: 0px 0px 0px 1rem;"
                  type="text"
                  value="14:00"
                />
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
